cfg/aws/aws_sdk_logging.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package aws
+
+import (
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/influxdata/wlog"
+)
+
+var sdkLogLevel aws.LogLevelType = aws.LogOff
+
+// SetSDKLogLevel sets the global log level which will be used in all AWS SDK calls.
+// Example usage: set `debug = true` and `aws_sdk_log_level = LogDebug` in config.json.
+func SetSDKLogLevel(wLogLevel wlog.Level, sdkLogLevelString string) {
+	if wLogLevel != wlog.DEBUG {
+		sdkLogLevel = aws.LogOff
+		return
+	}
```

Comment:
I expect customers to enable AWS SDK logging only when debugging an issue.
Therefore I expect customers to enable AWS SDK logging only when they enable CloudWatch debug level logging.
That is why I chose to log all messages from AWS SDK at a `D!` (debug) level in CloudWatch Agent.

I considered setting the AWS SDK log level to `LogDebugWithHTTPBody` anytime the Agent's JSON config has `"debug" : `true`. I decided against it because I think more control is better, by that I mean letting customers set the AWS SDK log level they want.

If I remove `if wLogLevel != wlog.DEBUG {` Then an Agent config like this would be doing wasted work of building AWS SDK log messages only to drop them.
```
{
    "agent": { 
        "debug": false, 
        "aws_sdk_log_level": "LogDebugWithRequestRetries"
    }
```

I prefer to keep the check `if wLogLevel != wlog.DEBUG {`.
Please let me know if you still think we should remove it.

Reply:
If we want to make the parameters independent, then I think we should make the AWS SDK messages show up even if cwagent `"debug"` is not set.  I don't see any way to do that other than to log them at `"I!"`, although I admit that's not awesome either.

Reply:
Okay, I will change to use `"I!"`.